disnake/member.py

```diff
@@ -960,3 +999,49 @@ def get_role(self, role_id: int, /) -> Optional[Role]:
             The role or ``None`` if not found in the member's roles.
         """
         return self.guild.get_role(role_id) if self._roles.has(role_id) else None
+
+    async def timeout(
+        self, seconds: Optional[int] = MISSING, reason: Optional[str] = None
+    ) -> Optional[Member]:
+        """|coro|
+
+        Time outs the member from the guild; until then, the member will not be able to interact with the guild.
+
+        You must have the :attr:`Permissions.moderate_members` permission to do this.
+
+        .. versionadded:: 2.3
+
+        Parameters
+        ----------
+        seconds: Optional[:class:`int`]
+            The seconds to time out. Set to ``None`` or ``0`` to remove the timeout.
+            Support up to ``2419200`` seconds (28 days) in the future.
+        reason: Optional[:class:`str`]
+            The reason for this time out. Appears on the audit log.
+
+        Raises
+        -------
+        Forbidden
+            You do not have permissions to time out this member.
+        HTTPException
+            Time outing the member failed.
+
+        Returns
+        -------
+        Optional[:class:`Member`]
+            The newly updated member.
+        """
+        http = self._state.http
+        guild_id = self.guild.id
+        payload: Dict[str, Any] = {}
+
+        if seconds is not MISSING:
+            if seconds and seconds > 0:
+                date_time = utils.utcnow() + datetime.timedelta(seconds=seconds)
+                payload["communication_disabled_until"] = date_time.isoformat()
+            else:
+                payload["communication_disabled_until"] = None
+
+        if payload:
+            data = await http.edit_member(guild_id, self.id, reason=reason, **payload)
+            return Member(data=data, guild=self.guild, state=self._state)
```

Comment:
To avoid repetitions, this can be simplified to `return await self.guild.timeout(self, seconds=seconds, reason=reason)`